packages/upload-media/src/components/provider/with-registry-provider.tsx

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+import { useRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
+// eslint-disable-next-line no-restricted-syntax
+import type { WPDataRegistry } from '@wordpress/data/build-types/registry';
+import { createHigherOrderComponent } from '@wordpress/compose';
+
+/**
+ * Internal dependencies
+ */
+import { storeConfig } from '../../store';
+import { STORE_NAME as mediaUploadStoreName } from '../../store/constants';
+
```

Comment:
The build-types directory should not be referenced directly. I think the registry type could probably just be exported from the data package, it's effectively public already.

If we don't want to export it here, it can be accessed in a better way like this:

```suggestion
import { useRegistry, createRegistry, RegistryProvider } from '@wordpress/data';
import { createHigherOrderComponent } from '@wordpress/compose';

/**
 * Internal dependencies
 */
import { storeConfig } from '../../store';
import { STORE_NAME as mediaUploadStoreName } from '../../store/constants';

type WPDataRegistry = ReturnType< typeof createRegistry >;

```